### PR TITLE
Use switch instead of if battery for improved performance

### PR DESCRIPTION
**Pull request by rubenporras**

### CODE REVIEW

**Review by joaodinissf — CHANGES_REQUESTED** [review]

File: com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend [hunk 1]
```diff
@@ -103,43 +103,52 @@ class ScopeProviderGenerator {
...
     }
 
     /** {@inheritDoc} */
     @Override
     protected IScope doGetScope(final EObject context, final EClass type, final String scopeName, final Resource originalResource) {
-      «FOR name : allScopes().filter(s|s.reference === null).map(s|s.getScopeName()).toSet().sortBy(n|if (n=="scope") "" else n) SEPARATOR " else "
-     »if ("«name»".equals(scopeName)) {
+      «FOR name : allScopes().filter(s|s.reference === null).map(s|s.getScopeName()).toSet()
+     »switch (scopeName) {
         «FOR scope : allScopes().filter(s|s.reference === null).filter(s|s.getScopeName()==name)»
```
[the remark this answers is not in the document]
Comment (rubenporras) [reply com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend:125]: It looks like that :) I also need to add a check that scopeName is not null before it is ready for review again.

File: com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend [hunk 2]
```diff
@@ -102,58 +102,77 @@ class ScopeProviderGenerator {
   def scopeMethods(ScopeModel it, String baseName) '''
     @Override
     protected IScope doGetScope(final EObject context, final EReference reference, final String scopeName, final Resource originalResource) {
-      «FOR name : allScopes().filter(s|s.reference !== null).map(s|s.getScopeName()).toSet().sortBy(n|if (n=="scope") "" else n) SEPARATOR " else "
-     »if ("«name»".equals(scopeName)) {
+      if (scopeName == null) {
+        return null;
+      }
+      switch (scopeName) {
```
Comment (joaodinissf) [inline com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend:108]:
```suggestion
      }

      switch (scopeName) {
```

For consistency with the other `doGetScope` below.

**Review by joaodinissf — DISMISSED** [review]
Minor suggestion above. Otherwise LGTM.

**Review by joaodinissf — DISMISSED** [review]

File: com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend [hunk 3]
```diff
@@ -102,58 +102,80 @@ class ScopeProviderGenerator {
...
+      «FOR name : allScopes().filter(s|s.reference !== null).map(s|s.getScopeName()).toSet()
+     »case "«name»":
         «FOR scope : allScopes().filter(s|s.reference !== null).filter(s|s.getScopeName()==name)»
         if (reference == «scope.reference.literalIdentifier()») return «scope.scopeMethodName()»(context, reference, originalResource);
         «ENDFOR»
-      }«
+        break;
+      «
       ENDFOR»
```
Comment (joaodinissf) [inline com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend:117]: Can these be combined into a single line, for consistency?

File: com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend [hunk 4]
```diff
@@ -102,58 +102,80 @@ class ScopeProviderGenerator {
...
+
+      switch (scopeName) {
+      «FOR name : allScopes().filter(s|s.reference === null).map(s|s.getScopeName()).toSet()
+     »case "«name»" :
         «FOR scope : allScopes().filter(s|s.reference === null).filter(s|s.getScopeName()==name)»
         if (type == «scope.targetType.literalIdentifier()») return «scope.scopeMethodName()»(context, type, originalResource);
         «ENDFOR»
-      }«
+        break;«
       ENDFOR»
```
Comment (joaodinissf) [inline com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend:136]: Can these be combined into a single line, for consistency?

File: com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend [hunk 5]
```diff
@@ -102,58 +102,80 @@ class ScopeProviderGenerator {
...
+       »case "«name»":
           «FOR scope : allScopes().filter(s|s.reference !== null).filter(s|s.getScopeName()==name)»
           «val globalRules = scope.allScopeRules().filter(r|r.context.global)»
           «IF globalRules.size > 0»
           if (reference == «scope.reference.literalIdentifier()») return true;
           «ENDIF»
           «ENDFOR»
-        }«
+          break;«
         ENDFOR»
```
Comment (joaodinissf) [inline com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend:155]: Can these be combined into a single line, for consistency?

File: com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend [hunk 6]
```diff
@@ -102,58 +102,80 @@ class ScopeProviderGenerator {
...
           «FOR scope : allScopes().filter(s|s.reference === null).filter(s|s.getScopeName()==name)»
           «val globalRules = scope.allScopeRules().filter(r|r.context.global)»
           «IF globalRules.size > 0»
           if (type == «scope.targetType.literalIdentifier()») return true;
           «ENDIF»
           «ENDFOR»
-        }«
+          break;
+        «
         ENDFOR»
```
Comment (joaodinissf) [inline com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend:176]: Can these be combined into a single line, for consistency?

**Review by joaodinissf — CHANGES_REQUESTED** [review]

File: com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend (hunk 3 above)
Comment (rubenporras) [reply com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend]: done

File: com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend (hunk 5 above)
Comment (rubenporras) [reply com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend]: done

File: com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend (hunk 6 above)
Comment (rubenporras) [reply com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend]: done

**Review by joaodinissf — DISMISSED** [review]

**Review by joaodinissf — APPROVED** [review]

**Pull request merged**